diff -ruN x-loader-v1.51_OMAPPSP_04.02.00.07.orig/board/omap3beagle/omap3beagle.c x-loader-v1.51_OMAPPSP_04.02.00.07/board/omap3beagle/omap3beagle.c
--- x-loader-v1.51_OMAPPSP_04.02.00.07.orig/board/omap3beagle/omap3beagle.c	2011-04-08 14:08:24.000000000 +0000
+++ x-loader-v1.51_OMAPPSP_04.02.00.07/board/omap3beagle/omap3beagle.c	2012-01-22 20:07:29.981018000 +0000
@@ -22,7 +22,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
  * MA 02111-1307 USA
  */
-
 #include <common.h>
 #include <command.h>
 #include <part.h>
@@ -69,7 +68,6 @@
 	unsigned int m2div;
 };
 
-
 typedef struct dpll_param dpll_param;
 
 /* Following functions are exported from lowlevel_init.S */
@@ -1112,6 +1111,8 @@
 	MUX_VAL(CP(UART1_TX),       (IDIS | PTD | DIS | M0)) /*UART1_TX*/\
 	MUX_VAL(CP(UART1_RTS),      (IDIS | PTD | DIS | M4)) /*GPIO_149*/\
 	MUX_VAL(CP(UART1_CTS),      (IDIS | PTD | DIS | M4)) /*GPIO_150*/\
+	MUX_VAL(CP(MMC2_DAT6),      (IDIS | PTD | DIS | M4)) /*GPIO_138*/\
+	MUX_VAL(CP(MMC2_DAT7),      (IDIS | PTD | DIS | M4)) /*GPIO_139*/\
 	MUX_VAL(CP(UART1_RX),       (IEN  | PTD | DIS | M0)) /*UART1_RX*/\
 	MUX_VAL(CP(UART3_CTS_RCTX), (IEN  | PTD | EN  | M0)) /*UART3_CTS_RCTX */\
 	MUX_VAL(CP(UART3_RTS_SD),   (IDIS | PTD | DIS | M0)) /*UART3_RTS_SD */\
@@ -1258,21 +1258,35 @@
 }
 
 typedef int (mmc_boot_addr) (void);
+void 	udelay (unsigned long usec);
 int mmc_boot(unsigned char *buf)
 {
 
 	long size = 0;
+	char i = 0;
 #ifdef CFG_CMD_FAT
 	block_dev_desc_t *dev_desc = NULL;
 	unsigned char ret = 0;
 
 	printf("Starting X-loader on MMC \n");
 
+	if (omap_request_gpio(139) == -22)
+		printf ("Erro requisitando GPIO\n");
+		
+	omap_set_gpio_direction(139,0);
+	//while(1) {
+		////printf("Teste FLUS %d\n", i);
+		//omap_set_gpio_dataout(139,i);
+		//i = (i + 1) & 1;
+		//delay(4320);
+	//}
+
 	ret = mmc_init(1);
 	if (ret == 0) {
 		printf("\n MMC init failed \n");
 		return 0;
 	}
+	
 
 	dev_desc = (block_dev_desc_t *)mmc_get_dev(0);
 	fat_register_device(dev_desc, 1);
